sdks/java/extensions/sketching/src/main/java/org/apache/beam/sdk/extensions/sketching/frequency/KMostFrequent.java

```diff
@@ -0,0 +1,200 @@
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.sketching.frequency;
+
+import com.clearspring.analytics.stream.Counter;
+import com.clearspring.analytics.stream.StreamSummary;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Iterator;
+import java.util.List;
+
+import org.apache.beam.sdk.coders.ByteArrayCoder;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.CoderRegistry;
+import org.apache.beam.sdk.coders.CustomCoder;
+import org.apache.beam.sdk.transforms.Combine;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@code PTransform}s for finding the k most frequent elements in a {@code PCollection}, or
+ * the k most frequent values associated with each key in a {@code PCollection} of {@code KV}s.
+ *
+ * <p>This class uses the Space-Saving algorithm, introduced in this paper :
+ * https://pdfs.semanticscholar.org/72f1/5aba2e67b1cc9cd1fb12c99e101c4c1aae4b.pdf
+ * <br>The implementation comes from Addthis' library Stream-lib : https://github.com/addthis/stream-lib
+ */
+public class KMostFrequent {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KMostFrequent.class);
+
+    // do not instantiate
+  private KMostFrequent() {
+  }
+
+  /**
+   * A {@code PTransform} that takes an inputT {@code PCollection} of objects and returns a
+   * {@code PCollection<StreamSummary>} whose contents is a sketch which contains
+   * the most frequent elements in the input {@code PCollection}.
+   *
+   * <p>The {@code capacity} parameter controls the maximum number of elements the sketch
+   * can contain. Once this capacity is reached, the least frequent element is dropped each
+   * time an incoming element is not already present in the sketch.
+   * Each element in the sketch is associated to a counter, that keeps track of the estimate
+   * frequency as well as the maximal potential error.
+   * <br>See {@link KMostFrequentFn} for more details.
+   *
+   * <p>Example of use
+   * <pre> {@code PCollection<String> input = ...;
+   * PCollection<StreamSummary<String>> ssSketch = input
+   *        .apply(SketchFrequencies.<String>perKey(10000));
```

Comment:
<!--new_thread; commit:6bf51fdf6afe6f389415792f335dd08adf9fa85a; resolved:0-->
Shouldn't the example be for KMostFrequent?